mlrun/artifacts/llm_prompt.py

```diff
@@ -0,0 +1,136 @@
+# Copyright 2025 Iguazio
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import tempfile
+import typing
+
+import mlrun
+import mlrun.artifacts.model as model_artifact
+from mlrun.artifacts import Artifact, ArtifactSpec
+from mlrun.utils import StorePrefix, logger
+
+MAX_PROMPT_LENGTH = 1000
+
+
+class LLMPromptArtifactSpec(ArtifactSpec):
+    _dict_fields = ArtifactSpec._dict_fields + [
+        "prompt_string",
+        "prompt_legend",
+        "generation_configuration",
+        "description",
+    ]
+
+    def __init__(
+        self,
+        target_path=None,
+        viewer=None,
+        is_inline=False,
+        format=None,
+        size=None,
+        db_key=None,
+        extra_data=None,
+        body=None,
+        unpackaging_instructions: typing.Optional[dict] = None,
+        parent_uri: typing.Optional[str] = None,
+        prompt_string: typing.Optional[str] = None,
+        prompt_path: typing.Optional[str] = None,
+        prompt_legend: typing.Optional[dict] = None,
+        generation_configuration: typing.Optional[dict] = None,
```

Comment:
I don't like the name "generation configuration" - this is not the config of the generation of the prompt, but rather this is expected to be the model configuration to be used. So why not call this `model_configuration`?